### PR TITLE
Refactoring core functionality

### PR DESCRIPTION
**Pull request by furiousdom**

### CODE REVIEW

File: bin/index.js [hunk 1]
```diff
@@ -2,43 +2,27 @@
...
-  const values = Object.values(element);
-  values.forEach((val, i) => {
-    csvCode += val;
-    const isLastValue = (values.length - 1) === i;
-    if (isLastValue) return;
-    csvCode += ',';
-  });
-  csvCode += '\n';
-});
+const csvCode = utils.createCsvCode(columns, values);
```
Comment (abasic) [inline bin/index.js:25]: Is there a reason why this also is not a function?

File: bin/utils.js [hunk 2]
```diff
@@ -1,16 +1,30 @@
...
+}
+
+function getAllValues(arr) {
+  return arr.map(getObjectValues);
+}
+
+function createCsvLine(arr) {
+  return arr.reduce((accumulator, currentValue) => {
+    return accumulator + ',' + currentValue;
+  });
```
Comment (abasic) [inline bin/utils.js:15]: How does this work when the starting value for `reduce` is not defined?

File: bin/utils.js [hunk 3]
```diff
@@ -1,16 +1,30 @@
...
+}
+
+function createCsvLines(csvValues) {
+  return csvValues.reduce((accumulator, currentValue) => {
```
Comment (abasic) [inline bin/utils.js:19]: How does this work when the starting value for `reduce` is not defined?

File: bin/utils.js [hunk 4]
```diff
@@ -1,16 +1,30 @@
...
-  else {
-    args.forEach((arg, i) => {
-      files.push(arg);
-    });
-  }
-  return files;
+function createCsvCode(csvColumns, csvValues) {
+  const csvColumnsText = createCsvLine(csvColumns);
+  const csvValuesText = createCsvLines(csvValues);
+  return csvColumnsText + '\n' + csvValuesText;
```
Comment (abasic) [inline bin/utils.js:27]: Can this be shorter somehow?

File: bin/index.js (hunk 1 above)
Comment (furiousdom) [reply bin/index.js:25]: There is no particular reason, other than at the time of writing this code this was the best solution that I could come up with. Next commit will make it more consistent.

**Review by abasic — APPROVED** [review]

**Pull request merged**